src/CosmosDB/CosmosDB/SQL/InvokeAzCosmosDBSqlContainerMerge.cs

```diff
@@ -0,0 +1,117 @@
+        [ValidateNotNullOrEmpty]
+        public string DatabaseName;
+
+        [Parameter(Mandatory = true, ParameterSetName = NameParameterSet, HelpMessage = Constants.ResourceGroupNameHelpMessage)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = Constants.ContainerNameHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }        
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = ParentObjectParameterSet, HelpMessage = Constants.SqlDatabaseObjectHelpMessage)]
+        [ValidateNotNull]
+        public PSSqlDatabaseGetResults ParentObject { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = ObjectParameterSet, HelpMessage = Constants.SqlContainerObjectHelpMessage)]
+        [ValidateNotNull]
+        public PSSqlContainerGetResults InputObject { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter Force { get; set; }
+
+        public void PopulateFromParentObject()
+        {
+            ResourceIdentifier resourceIdentifier = new ResourceIdentifier(ParentObject.Id);
+            ResourceGroupName = resourceIdentifier.ResourceGroupName;
+            DatabaseName = resourceIdentifier.ResourceName;
+            AccountName = ResourceIdentifierExtensions.GetDatabaseAccountName(resourceIdentifier);
+        }
+
+        public void PopulateFromInputObject()
+        {
+            ResourceIdentifier resourceIdentifier = new ResourceIdentifier(InputObject.Id);
+            ResourceGroupName = resourceIdentifier.ResourceGroupName;
+            Name = resourceIdentifier.ResourceName;
+            DatabaseName = ResourceIdentifierExtensions.GetSqlDatabaseName(resourceIdentifier);
+            AccountName = ResourceIdentifierExtensions.GetDatabaseAccountName(resourceIdentifier);
+        }
+
+        public override void ExecuteCmdlet()
+        {
+            if (ParameterSetName.Equals(ParentObjectParameterSet, StringComparison.Ordinal))
+            {
+                PopulateFromParentObject();
+            }
+            else if (ParameterSetName.Equals(ObjectParameterSet, StringComparison.Ordinal))
+            {
+                PopulateFromInputObject();
+            }
+
+            PhysicalPartitionStorageInfoCollection physicalPartitionStorageInfoCollection = null;
+            MergeParameters mergeParameters = new MergeParameters(isDryRun: false);
+
+            if (ShouldProcess(Name, "Merging partitions.", String.Empty, out ShouldProcessReason shouldProcessReason))
+            {
+                if (this.Force.IsPresent || ShouldContinue($"This command will merge the partitions of collection {Name} , do you want to continue?", String.Empty))
+                {
+                    physicalPartitionStorageInfoCollection =
```

Comment:
can we add a message here like -" this operation might take a long time (potentially hours) depending on the data movement involved "